### ISSUE TITLE
remove BoundFieldAccess.IsByValue

### ISSUE THREAD

**Issue by VSadov** [dotnet/roslyn#21265]
We have a more comprehensive mechanism to track fields that cannot be used by reference. 

There is no need to have this now. It should use the same infrastructure or we will keep forgetting about this.

**Comment by VSadov** [comment]
Also consider thsi bizzare behavior enabled byt this flag:

```cs
    class Program
    {
        static void Main(string[] args)
        {
            S1[] a = new S1[1];

            foreach(var t in a)
            {
                System.Console.WriteLine(t.y.x);
                t.y.Increment();
                System.Console.WriteLine(t.y.x);

                System.Console.WriteLine(t.x);
                t.Increment();
                System.Console.WriteLine(t.x);
            }
        }
    }

    struct S1
    {
        public int x;

        public S2 y;

        public void Increment()
        {
            x++;
        }
    }

    struct S2
    {
        public int x;

        public void Increment()
        {
            x++;
        }
    }
```

prints   
```
0
0
0
1
```

Really? 
Is `t` readonly or not?

**Comment by VSadov** [comment]
I think this needs further discussion. Too risky for 15.7
